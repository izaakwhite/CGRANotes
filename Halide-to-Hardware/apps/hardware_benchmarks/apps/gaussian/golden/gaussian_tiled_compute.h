#pragma once
#include "hw_classes.h"
#include "clockwork_standard_compute_units.h"


//store is: hw_input_global_wrapper.stencil(hw_input_global_wrapper_s0_x, hw_input_global_wrapper_s0_y) = hw_input.stencil(hw_input_global_wrapper_s0_x, hw_input_global_wrapper_s0_y)
hw_uint<16> hcompute_hw_input_global_wrapper_stencil(hw_uint<16>& hw_input_stencil) {
  uint16_t _hw_input_stencil_1 = (uint16_t) hw_input_stencil.extract<0, 15>();

  return _hw_input_stencil_1;
}

//store is: blur_unnormalized.stencil(blur_unnormalized_s0_x, blur_unnormalized_s0_y) = (uint16)0
hw_uint<16> hcompute_blur_unnormalized_stencil() {
  uint16_t _274 = (uint16_t)(0);
  return _274;
}

//store is: blur_unnormalized.stencil(blur_unnormalized_s1_x, blur_unnormalized_s1_y) = ((hw_input_global_wrapper.stencil(blur_unnormalized_s1_x, blur_unnormalized_s1_y)*(uint16)24) + (blur_unnormalized.stencil(blur_unnormalized_s1_x, blur_unnormalized_s1_y) + ((hw_input_global_wrapper.stencil((blur_unnormalized_s1_x + 1), blur_unnormalized_s1_y)*(uint16)30) + ((hw_input_global_wrapper.stencil((blur_unnormalized_s1_x + 2), blur_unnormalized_s1_y)*(uint16)24) + ((hw_input_global_wrapper.stencil(blur_unnormalized_s1_x, (blur_unnormalized_s1_y + 1))*(uint16)30) + ((hw_input_global_wrapper.stencil((blur_unnormalized_s1_x + 1), (blur_unnormalized_s1_y + 1))*(uint16)37) + ((hw_input_global_wrapper.stencil((blur_unnormalized_s1_x + 2), (blur_unnormalized_s1_y + 1))*(uint16)30) + ((hw_input_global_wrapper.stencil(blur_unnormalized_s1_x, (blur_unnormalized_s1_y + 2))*(uint16)24) + ((hw_input_global_wrapper.stencil((blur_unnormalized_s1_x + 2), (blur_unnormalized_s1_y + 2))*(uint16)24) + (hw_input_global_wrapper.stencil((blur_unnormalized_s1_x + 1), (blur_unnormalized_s1_y + 2))*(uint16)30))))))))))
hw_uint<16> hcompute_blur_unnormalized_stencil_1(hw_uint<16>& blur_unnormalized_stencil, hw_uint<144>& hw_input_global_wrapper_stencil) {
  uint16_t _blur_unnormalized_stencil_1 = (uint16_t) blur_unnormalized_stencil.extract<0, 15>();

  uint16_t _hw_input_global_wrapper_stencil_1 = (uint16_t) hw_input_global_wrapper_stencil.extract<0, 15>();
  uint16_t _hw_input_global_wrapper_stencil_2 = (uint16_t) hw_input_global_wrapper_stencil.extract<16, 31>();
  uint16_t _hw_input_global_wrapper_stencil_3 = (uint16_t) hw_input_global_wrapper_stencil.extract<32, 47>();
  uint16_t _hw_input_global_wrapper_stencil_4 = (uint16_t) hw_input_global_wrapper_stencil.extract<48, 63>();
  uint16_t _hw_input_global_wrapper_stencil_5 = (uint16_t) hw_input_global_wrapper_stencil.extract<64, 79>();
  uint16_t _hw_input_global_wrapper_stencil_6 = (uint16_t) hw_input_global_wrapper_stencil.extract<80, 95>();
  uint16_t _hw_input_global_wrapper_stencil_7 = (uint16_t) hw_input_global_wrapper_stencil.extract<96, 111>();
  uint16_t _hw_input_global_wrapper_stencil_8 = (uint16_t) hw_input_global_wrapper_stencil.extract<112, 127>();
  uint16_t _hw_input_global_wrapper_stencil_9 = (uint16_t) hw_input_global_wrapper_stencil.extract<128, 143>();

  uint16_t _277 = (uint16_t)(24);
  uint16_t _278 = _hw_input_global_wrapper_stencil_1 * _277;
  uint16_t _279 = (uint16_t)(30);
  uint16_t _280 = _hw_input_global_wrapper_stencil_2 * _279;
  uint16_t _281 = _hw_input_global_wrapper_stencil_3 * _277;
  uint16_t _282 = _hw_input_global_wrapper_stencil_4 * _279;
  uint16_t _283 = (uint16_t)(37);
  uint16_t _284 = _hw_input_global_wrapper_stencil_5 * _283;
  uint16_t _285 = _hw_input_global_wrapper_stencil_6 * _279;
  uint16_t _286 = _hw_input_global_wrapper_stencil_7 * _277;
  uint16_t _287 = _hw_input_global_wrapper_stencil_8 * _277;
  uint16_t _288 = _hw_input_global_wrapper_stencil_9 * _279;
  uint16_t _289 = _287 + _288;
  uint16_t _290 = _286 + _289;
  uint16_t _291 = _285 + _290;
  uint16_t _292 = _284 + _291;
  uint16_t _293 = _282 + _292;
  uint16_t _294 = _281 + _293;
  uint16_t _295 = _280 + _294;
  uint16_t _296 = _blur_unnormalized_stencil_1 + _295;
  uint16_t _297 = _278 + _296;
  return _297;
}

//store is: blur.stencil(blur_s0_x, blur_s0_y) = (blur_unnormalized.stencil(blur_s0_x, blur_s0_y)/(uint16)256)
hw_uint<16> hcompute_blur_stencil(hw_uint<16>& blur_unnormalized_stencil) {
  uint16_t _blur_unnormalized_stencil_2 = (uint16_t) blur_unnormalized_stencil.extract<0, 15>();

  uint16_t _354 = (uint16_t)(8);
  uint16_t _355 = _blur_unnormalized_stencil_2 >> _354;
  return _355;
}

//store is: hw_output.stencil(hw_output_s0_x_xi, hw_output_s0_y_yi) = blur.stencil(hw_output_s0_x_xi, hw_output_s0_y_yi)
hw_uint<16> hcompute_hw_output_stencil(hw_uint<16>& blur_stencil) {
  uint16_t _blur_stencil_1 = (uint16_t) blur_stencil.extract<0, 15>();

  return _blur_stencil_1;
}

