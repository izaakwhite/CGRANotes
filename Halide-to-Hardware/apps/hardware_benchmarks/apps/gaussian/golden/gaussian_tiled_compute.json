{
"namespaces":{
  "global":{
    "modules":{
      "hcompute_blur_stencil":{
        "type":["Record",[
          ["out_blur_stencil",["Array",16,"Bit"]],
          ["in0_blur_unnormalized_stencil",["Array",1,["Array",16,"BitIn"]]]
        ]],
        "instances":{
          "const_p8__356":{
            "genref":"coreir.const",
            "genargs":{"width":["Int",16]},
            "modargs":{"value":[["BitVector",16],"16'h0008"]}
          },
          "lshr_blur_unnormalized_stencil_2_356_357":{
            "genref":"coreir.lshr",
            "genargs":{"width":["Int",16]}
          }
        },
        "connections":[
          ["lshr_blur_unnormalized_stencil_2_356_357.in1","const_p8__356.out"],
          ["self.in0_blur_unnormalized_stencil.0","lshr_blur_unnormalized_stencil_2_356_357.in0"],
          ["self.out_blur_stencil","lshr_blur_unnormalized_stencil_2_356_357.out"]
        ]
      },
      "hcompute_blur_unnormalized_stencil":{
        "type":["Record",[
          ["out_blur_unnormalized_stencil",["Array",16,"Bit"]]
        ]],
        "instances":{
          "const_p0__275":{
            "genref":"coreir.const",
            "genargs":{"width":["Int",16]},
            "modargs":{"value":[["BitVector",16],"16'h0000"]}
          }
        },
        "connections":[
          ["self.out_blur_unnormalized_stencil","const_p0__275.out"]
        ]
      },
      "hcompute_blur_unnormalized_stencil_1":{
        "type":["Record",[
          ["out_blur_unnormalized_stencil",["Array",16,"Bit"]],
          ["in0_blur_unnormalized_stencil",["Array",1,["Array",16,"BitIn"]]],
          ["in1_hw_input_global_wrapper_stencil",["Array",9,["Array",16,"BitIn"]]]
        ]],
        "instances":{
          "add_299_317_318":{
            "genref":"coreir.add",
            "genargs":{"width":["Int",16]}
          },
          "add_301_315_316":{
            "genref":"coreir.add",
            "genargs":{"width":["Int",16]}
          },
          "add_302_314_315":{
            "genref":"coreir.add",
            "genargs":{"width":["Int",16]}
          },
          "add_303_313_314":{
            "genref":"coreir.add",
            "genargs":{"width":["Int",16]}
          },
          "add_305_312_313":{
            "genref":"coreir.add",
            "genargs":{"width":["Int",16]}
          },
          "add_306_311_312":{
            "genref":"coreir.add",
            "genargs":{"width":["Int",16]}
          },
          "add_307_310_311":{
            "genref":"coreir.add",
            "genargs":{"width":["Int",16]}
          },
          "add_308_309_310":{
            "genref":"coreir.add",
            "genargs":{"width":["Int",16]}
          },
          "add_blur_unnormalized_stencil_1_316_317":{
            "genref":"coreir.add",
            "genargs":{"width":["Int",16]}
          },
          "const_p24__298":{
            "genref":"coreir.const",
            "genargs":{"width":["Int",16]},
            "modargs":{"value":[["BitVector",16],"16'h0018"]}
          },
          "const_p24__298$1":{
            "genref":"coreir.const",
            "genargs":{"width":["Int",16]},
            "modargs":{"value":[["BitVector",16],"16'h0018"]}
          },
          "const_p24__298$2":{
            "genref":"coreir.const",
            "genargs":{"width":["Int",16]},
            "modargs":{"value":[["BitVector",16],"16'h0018"]}
          },
          "const_p24__298$3":{
            "genref":"coreir.const",
            "genargs":{"width":["Int",16]},
            "modargs":{"value":[["BitVector",16],"16'h0018"]}
          },
          "const_p30__300":{
            "genref":"coreir.const",
            "genargs":{"width":["Int",16]},
            "modargs":{"value":[["BitVector",16],"16'h001e"]}
          },
          "const_p30__300$1":{
            "genref":"coreir.const",
            "genargs":{"width":["Int",16]},
            "modargs":{"value":[["BitVector",16],"16'h001e"]}
          },
          "const_p30__300$2":{
            "genref":"coreir.const",
            "genargs":{"width":["Int",16]},
            "modargs":{"value":[["BitVector",16],"16'h001e"]}
          },
          "const_p30__300$3":{
            "genref":"coreir.const",
            "genargs":{"width":["Int",16]},
            "modargs":{"value":[["BitVector",16],"16'h001e"]}
          },
          "const_p37__304":{
            "genref":"coreir.const",
            "genargs":{"width":["Int",16]},
            "modargs":{"value":[["BitVector",16],"16'h0025"]}
          },
          "mul_hw_input_global_wrapper_stencil_1_298_299":{
            "genref":"coreir.mul",
            "genargs":{"width":["Int",16]}
          },
          "mul_hw_input_global_wrapper_stencil_2_300_301":{
            "genref":"coreir.mul",
            "genargs":{"width":["Int",16]}
          },
          "mul_hw_input_global_wrapper_stencil_3_298_302":{
            "genref":"coreir.mul",
            "genargs":{"width":["Int",16]}
          },
          "mul_hw_input_global_wrapper_stencil_4_300_303":{
            "genref":"coreir.mul",
            "genargs":{"width":["Int",16]}
          },
          "mul_hw_input_global_wrapper_stencil_5_304_305":{
            "genref":"coreir.mul",
            "genargs":{"width":["Int",16]}
          },
          "mul_hw_input_global_wrapper_stencil_6_300_306":{
            "genref":"coreir.mul",
            "genargs":{"width":["Int",16]}
          },
          "mul_hw_input_global_wrapper_stencil_7_298_307":{
            "genref":"coreir.mul",
            "genargs":{"width":["Int",16]}
          },
          "mul_hw_input_global_wrapper_stencil_8_298_308":{
            "genref":"coreir.mul",
            "genargs":{"width":["Int",16]}
          },
          "mul_hw_input_global_wrapper_stencil_9_300_309":{
            "genref":"coreir.mul",
            "genargs":{"width":["Int",16]}
          }
        },
        "connections":[
          ["mul_hw_input_global_wrapper_stencil_1_298_299.out","add_299_317_318.in0"],
          ["add_blur_unnormalized_stencil_1_316_317.out","add_299_317_318.in1"],
          ["self.out_blur_unnormalized_stencil","add_299_317_318.out"],
          ["mul_hw_input_global_wrapper_stencil_2_300_301.out","add_301_315_316.in0"],
          ["add_302_314_315.out","add_301_315_316.in1"],
          ["add_blur_unnormalized_stencil_1_316_317.in1","add_301_315_316.out"],
          ["mul_hw_input_global_wrapper_stencil_3_298_302.out","add_302_314_315.in0"],
          ["add_303_313_314.out","add_302_314_315.in1"],
          ["mul_hw_input_global_wrapper_stencil_4_300_303.out","add_303_313_314.in0"],
          ["add_305_312_313.out","add_303_313_314.in1"],
          ["mul_hw_input_global_wrapper_stencil_5_304_305.out","add_305_312_313.in0"],
          ["add_306_311_312.out","add_305_312_313.in1"],
          ["mul_hw_input_global_wrapper_stencil_6_300_306.out","add_306_311_312.in0"],
          ["add_307_310_311.out","add_306_311_312.in1"],
          ["mul_hw_input_global_wrapper_stencil_7_298_307.out","add_307_310_311.in0"],
          ["add_308_309_310.out","add_307_310_311.in1"],
          ["mul_hw_input_global_wrapper_stencil_8_298_308.out","add_308_309_310.in0"],
          ["mul_hw_input_global_wrapper_stencil_9_300_309.out","add_308_309_310.in1"],
          ["self.in0_blur_unnormalized_stencil.0","add_blur_unnormalized_stencil_1_316_317.in0"],
          ["mul_hw_input_global_wrapper_stencil_3_298_302.in1","const_p24__298$1.out"],
          ["mul_hw_input_global_wrapper_stencil_7_298_307.in1","const_p24__298$2.out"],
          ["mul_hw_input_global_wrapper_stencil_8_298_308.in1","const_p24__298$3.out"],
          ["mul_hw_input_global_wrapper_stencil_1_298_299.in1","const_p24__298.out"],
          ["mul_hw_input_global_wrapper_stencil_4_300_303.in1","const_p30__300$1.out"],
          ["mul_hw_input_global_wrapper_stencil_6_300_306.in1","const_p30__300$2.out"],
          ["mul_hw_input_global_wrapper_stencil_9_300_309.in1","const_p30__300$3.out"],
          ["mul_hw_input_global_wrapper_stencil_2_300_301.in1","const_p30__300.out"],
          ["mul_hw_input_global_wrapper_stencil_5_304_305.in1","const_p37__304.out"],
          ["self.in1_hw_input_global_wrapper_stencil.0","mul_hw_input_global_wrapper_stencil_1_298_299.in0"],
          ["self.in1_hw_input_global_wrapper_stencil.1","mul_hw_input_global_wrapper_stencil_2_300_301.in0"],
          ["self.in1_hw_input_global_wrapper_stencil.2","mul_hw_input_global_wrapper_stencil_3_298_302.in0"],
          ["self.in1_hw_input_global_wrapper_stencil.3","mul_hw_input_global_wrapper_stencil_4_300_303.in0"],
          ["self.in1_hw_input_global_wrapper_stencil.4","mul_hw_input_global_wrapper_stencil_5_304_305.in0"],
          ["self.in1_hw_input_global_wrapper_stencil.5","mul_hw_input_global_wrapper_stencil_6_300_306.in0"],
          ["self.in1_hw_input_global_wrapper_stencil.6","mul_hw_input_global_wrapper_stencil_7_298_307.in0"],
          ["self.in1_hw_input_global_wrapper_stencil.7","mul_hw_input_global_wrapper_stencil_8_298_308.in0"],
          ["self.in1_hw_input_global_wrapper_stencil.8","mul_hw_input_global_wrapper_stencil_9_300_309.in0"]
        ]
      },
      "hcompute_hw_input_global_wrapper_stencil":{
        "type":["Record",[
          ["out_hw_input_global_wrapper_stencil",["Array",16,"Bit"]],
          ["in0_hw_input_stencil",["Array",1,["Array",16,"BitIn"]]]
        ]],
        "connections":[
          ["self.out_hw_input_global_wrapper_stencil","self.in0_hw_input_stencil.0"]
        ]
      },
      "hcompute_hw_output_stencil":{
        "type":["Record",[
          ["out_hw_output_stencil",["Array",16,"Bit"]],
          ["in0_blur_stencil",["Array",1,["Array",16,"BitIn"]]]
        ]],
        "connections":[
          ["self.out_hw_output_stencil","self.in0_blur_stencil.0"]
        ]
      }
    }
  }
}
}
